Fix some xiaomi device-specific changes (Mi Max 3, Jasmine, Wayne).

tytydraco opened the pull request:
1. Fix redundant device checks
2. Fix an old change that is no longer needed (iaomi --> xiaomi)
3. Fix front-facing flash on A2 / 6X
Signed-off-by: Tyler Nijmeh <tylernij@gmail.com>

tytydraco committed:
Fix typo with checking for the Mi Mix 3

There's a missing 'x' in 'xiaomi'. Initially, this was done before
`grep` was passed with the -i flag, which makes it useless now.

Signed-off-by: Tyler Nijmeh <tylernij@gmail.com>

phhusson commented on rw-system.sh:
```diff
@@ -203,6 +203,7 @@ if getprop ro.vendor.build.fingerprint | grep -q -i -e xiaomi/wayne -e xiaomi/ja
     setprop persist.imx376_sunny.light.lux 280
     setprop persist.imx376_ofilm.low.lux 310
     setprop persist.imx376_ofilm.light.lux 280
+    setprop ro.sf.lcd_density 460
```
It is not defined by vendor?
I haven't ever seen anyone complaining about this..?

tytydraco replied:
Nah basically on the Mi A2 / Mi 6X, the Quick Settings toggles are like right up against the sides of the panel. It's not right. I'll include a picture of before and after this change:

https://photos.app.goo.gl/kJasoSBxwSXzF22n6

phhusson commented on rw-system.sh:
```diff
@@ -69,7 +69,7 @@ changeKeylayout() {
         -e xiaomi/polaris -e xiaomi/sirius -e xiaomi/dipper \
         -e xiaomi/wayne -e xiaomi/jasmine -e xiaomi/jasmine_sprout \
-        -e xiaomi/platina -e iaomi/perseus -e xiaomi/ysl \
+        -e xiaomi/platina -e xiaomi/perseus -e xiaomi/ysl \
```
that's not a typo, but whatever

tytydraco replied:
iaomi isn't a typo? Ew

suwakowww replied:
because `Xiaomi` and `xiaomi`, seems specific for BSP devices and ODM devices, and `grep` just match keywords, and the keyword `iaomi` is not affected for matching device.

tytydraco replied:
You're passing -i to it in the beginning.

phhusson commented on rw-system.sh:
```diff
@@ -198,11 +198,27 @@ if [ "$(getprop ro.vendor.product.manufacturer)" = "motorola" ] || [ "$(getprop
 fi
 
-if getprop ro.vendor.build.fingerprint | grep -q -i -e xiaomi/wayne -e xiaomi/jasmine; then
+if getprop ro.vendor.build.fingerprint | grep -q -i -e xiaomi/wayne -e xiaomi/jasmine -e xiaomi/jasmine_sprout; then
```
jasmine matches jasmin_prout, so is it really useful?

tytydraco replied:
Ig not but it was defined earlier. Either remove both checks for _sprout or make them conform.

phhusson commented on rw-system.sh:
```diff
@@ -198,11 +198,27 @@ if [ "$(getprop ro.vendor.product.manufacturer)" = "motorola" ] || [ "$(getprop
+    # VoLTE
+    setprop persist.dbg.volte_avail_ovr 1
+
+    # Dual Camera
```
What's the issue you're trying to fix?

tytydraco replied:
Looking back, this sector of tweaks isn't necessary.

phhusson commented on rw-system.sh:
```diff
@@ -198,11 +198,27 @@ if [ "$(getprop ro.vendor.product.manufacturer)" = "motorola" ] || [ "$(getprop
     setprop persist.imx376_ofilm.light.lux 280
+    setprop ro.sf.lcd_density 460
+
+    # VoLTE
```
What is the issue you're trying to fix?
What's the behaviour with/without ims.apk without this prop, and with this prop?

tytydraco replied:
After looking, VoLTE works without this tweak, and so does the Dual Camera. Don't merge that last block, except for the lcd_density.

tytydraco committed:
Remove duplicate check for jasmine

Grepping for jasmine will also match jasmine_sprout.

Signed-off-by: Tyler Nijmeh <tylernij@gmail.com>

tytydraco committed:
Fix the front-facing flash on jasmine and wayne

For some reason, the front-facing flash stays enabled during camera
usage (front-facing only). This fixes that, while still keeping the
flash working.

Test: Ran the command and observed that front facing flash was no
longer stuck on.
Test: Ran the command and observed that flash still works when taking
photos with the flash enabled.

Signed-off-by: Tyler Nijmeh <tylernij@gmail.com>

tytydraco committed:
Fix Quick Settings icons being padded wrong

Signed-off-by: Tyler Nijmeh <tylernij@gmail.com>
